### PR TITLE
KOGITO-3818: Add Desktop integration tests

### PR DESCRIPTION
**Pull request by tomasdavidorg**
https://issues.redhat.com/browse/KOGITO-3818

### CODE REVIEW

File: packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap [hunk 1]
```diff
@@ -46,7 +46,7 @@ exports[`invalid file type alert alert is closed after 3000ms 1`] = `
           aria-disabled="false"
           aria-label="Close Danger alert: alert: This file extension is not supported."
           class="pf-c-button pf-m-plain"
-          data-ouia-component-id="OUIA-Generated-Button-plain-3"
+          data-ouia-component-id="OUIA-Generated-Button-plain-2"
```
Comment (jstastny-cz) [inline packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap]: This change should be harmless - the generated ids should be stable unless a new instance of the same component is added. And then snapshot would change anyway.

**Review by jstastny-cz — APPROVED** [review]
LGTM, just noticed a few generated OUIA ids - it seemed that you're trying to provide ids wherever possible, so you might wanna check those few.

File: packages/desktop/it-tests/NavigationTest.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { Application, SpectronClient } from "spectron";
+import { initApp } from "./Tools";
+
+let app: Application;
+let client: SpectronClient;
+
+beforeEach(async () => {
```
Comment (domhanak) [inline packages/desktop/it-tests/NavigationTest.ts:23]: I try to follow following format:

// before methods

// after methods

// tests

Do you mind changing this? :D For readability of course.

**Review by domhanak — APPROVED** [review]
small nitpick noted, otherwise LGTM.

File: packages/desktop/it-tests/NavigationTest.ts (hunk 2 above)
Comment (tomasdavidorg) [reply packages/desktop/it-tests/NavigationTest.ts:23]: Fixed.

File: packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap [hunk 3]
```diff
@@ -606,7 +608,7 @@ exports[`invalid file type alert alert is closed immediately after leaving the h
             aria-expanded="true"
             aria-label="Global navigation"
             class="pf-c-button pf-m-plain"
-            data-ouia-component-id="OUIA-Generated-Button-plain-4"
+            data-ouia-component-id="OUIA-Generated-Button-plain-2"
```
Comment (tomasdavidorg) [inline packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap:614]: @jstastny-cz This is the only change. Unfortunately, the toggle button is part of `PageHeader` and its `ouiaId` attribute cannot be changed (I didn't find a way how to do it.)

File: packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap (hunk 1 above)
Comment (tomasdavidorg) [reply packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap]: This one was fixed but another one appeared. See my comment below.

**Comment by tomasdavidorg** [comment]
@tiagobento Could you please take a look or assign to someone for a review? Thanks.

File: packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap (hunk 3 above)
Comment (jstastny-cz) [reply packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap:614]: ok, that's a common pitfall. Thanks.

File: packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap (hunk 1 above)
Comment (jstastny-cz) [reply packages/desktop/src/__tests__/webview/__snapshots__/App.test.tsx.snap]: Ack, thanks for looking into it again. Good to go.

**Comment by tiagobento** [comment]
Is this good to review?

**Review by tiagobento — APPROVED** [review]

**Comment by tomasdavidorg** [comment]
@tiagobento please merge if it is ok from eng. side. Thanks 🙂

File: .github/workflows/it-tests.yml [hunk 4]
```diff
@@ -119,6 +119,46 @@ jobs:
...
+      - uses: actions/checkout@v2
+      - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+
+      - name: Setup Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Setup Yarn
+        run: npm install -g yarn@${{ matrix.yarn }}
```
Comment (tiagobento) [inline .github/workflows/it-tests.yml]: Can you please add the network timeout here? Otherwise we'll see intermittent failures.

Comment (tomasdavidorg) [reply .github/workflows/it-tests.yml]: Added. Thanks.

**Comment by tomasdavidorg** [comment]
@tiagobento Could you please merge?

**Comment by tiagobento** [comment]
@tomasdavidorg There are some conflicts on this PR, can you please take a look?

**Comment by tomasdavidorg** [comment]
@tiagobento I rebased this with master branch + I updated electron version to `^11.4.1`. Lets wait for the checks

**Review by tiagobento — APPROVED** [review]

**Pull request merged**